server/src/main/java/prompt/ls1/controller/ApplicationController.java

```diff
@@ -0,0 +1,167 @@
+            @RequestParam(required = false, defaultValue = "false") boolean accepted
+    ) {
+        final CourseIteration courseIteration = courseIterationService.findBySemesterName(courseIterationName);
+
+        return ResponseEntity.ok(applicationService.findAllTutorApplicationsByCourseIteration(courseIteration.getId(), accepted));
+    }
+
+    @PostMapping("/developer")
+    public ResponseEntity<Application> createDeveloperApplication(@RequestBody DeveloperApplication developerApplication,
+                                              @RequestParam(name = "courseIteration") String courseIterationName) {
+        if (bucket.tryConsume(1)) {
+            final CourseIteration courseIteration = courseIterationService.findBySemesterName(courseIterationName);
+            developerApplication.setCourseIteration(courseIteration);
+
+            return ResponseEntity.ok(applicationService.createDeveloperApplication(developerApplication));
+        }
+
+        return ResponseEntity.status(HttpStatus.TOO_MANY_REQUESTS).build();
+    }
+
+    @PostMapping("/tutor")
+    public ResponseEntity<Application> createTutorApplication(@RequestBody TutorApplication tutorApplication,
+                                                                  @RequestParam(name = "courseIteration") String courseIterationName) {
+        if (bucket.tryConsume(1)) {
+            final CourseIteration courseIteration = courseIterationService.findBySemesterName(courseIterationName);
+            tutorApplication.setCourseIteration(courseIteration);
+
+            return ResponseEntity.ok(applicationService.createTutorApplication(tutorApplication));
+        }
+
+        return ResponseEntity.status(HttpStatus.TOO_MANY_REQUESTS).build();
+    }
+
+    @PostMapping("/coach")
+    public ResponseEntity<Application> createCoachApplication(@RequestBody CoachApplication coachApplication,
+                                                                  @RequestParam(name = "courseIteration") String courseIterationName) {
+        if (bucket.tryConsume(1)) {
+            final CourseIteration courseIteration = courseIterationService.findBySemesterName(courseIterationName);
+            coachApplication.setCourseIteration(courseIteration);
+
+            return ResponseEntity.ok(applicationService.createCoachApplication(coachApplication));
+        }
+
+        return ResponseEntity.status(HttpStatus.TOO_MANY_REQUESTS).build();
+    }
+
+    @PatchMapping(path = "/developer/{developerApplicationId}", consumes = "application/json-path+json")
+    public ResponseEntity<Application> updateProjectTeam(@PathVariable final UUID developerApplicationId,
+                                                         @RequestBody JsonPatch patchStudentApplication)
+            throws JsonPatchException, JsonProcessingException {
+        return ResponseEntity.ok(applicationService.updateDeveloperApplication(developerApplicationId, patchStudentApplication));
+    }
```

Comment:
Why are only the POST requests rate limited and not the PATCH ones? Couldn't I ddos by updating my application frequently?